base/sparse/sparsematrix.jl

```diff
@@ -2232,7 +2234,9 @@ getindex{T<:Integer}(A::SparseMatrixCSC, I::AbstractVector{Bool}, J::AbstractVec
 function setindex!{T,Ti}(A::SparseMatrixCSC{T,Ti}, v, i0::Integer, i1::Integer)
     i0 = convert(Ti, i0)
     i1 = convert(Ti, i1)
-    if !(1 <= i0 <= A.m && 1 <= i1 <= A.n); throw(BoundsError()); end
+    if !(1 <= i0 <= A.m && 1 <= i1 <= A.n)
+        throw(BoundsError())
```

Comment:
For many of the sparse indexing functions I would guess an extra GC frame would be insignificant but if it can be taken away easily, why not.